builder/dockerfile/internals.go

```diff
@@ -137,6 +171,53 @@ func (b *Builder) performCopy(state *dispatchState, inst copyInstruction) error
 	return b.exportImage(state, imageMount, runConfigWithCommentCmd)
 }
 
+func parseChownFlag(flChown string) (string, string) {
+	parts := strings.Split(flChown, ":")
+	if len(parts) == 1 {
+		// if no group specified, use the user spec as group as well
+		return parts[0], parts[0]
+	}
+	return parts[0], parts[1]
+}
+
+func lookupUser(userStr, filepath string) (int, error) {
+	// if the string is actually a uid integer, parse to int and return
+	// as we don't need to translate with the help of files
+	uid, err := strconv.Atoi(userStr)
+	if err == nil {
+		return uid, nil
+	}
+	users, err := lcUser.ParsePasswdFileFilter(filepath, func(u lcUser.User) bool {
+		if u.Name == userStr {
+			return true
+		}
+		return false
+	})
+	if err != nil {
+		return -1, err
```

Comment:
Minor thing, but it is kind of odd returning `-1` here, as `-1` in chown specifically means "do not chown this part" eg so you can chown a user but not a group, but you are not actually using that, so returning `0` ie the empty `int` is probably clearer. I had to double check you were not using the value regardless of the error code...